Integration of Mulitplicity Histos

makor opened the pull request:
Also sign fix in  PWGGA/Hyperon/macros/AddTaskSigma0Run2syst.C.

a-mathis commented on PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx:
```diff
@@ -55,11 +55,13 @@ ClassImp(AliSigma0PhotonMotherCuts)
+      fHistPtMult(),
       fHistArmenterosBefore(nullptr),
       fHistArmenterosAfter(nullptr),
       fHistMixedPtY(),
```
```suggestion
       fHistMixedPtY(),
      fHistMixedPtMult(),
```

a-mathis commented on PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx:
```diff
@@ -134,11 +136,13 @@ AliSigma0PhotonMotherCuts::AliSigma0PhotonMotherCuts(
+      fHistPtMult(),
       fHistArmenterosBefore(nullptr),
       fHistArmenterosAfter(nullptr),
       fHistMixedPtY(),
```
```suggestion
      fHistMixedPtY(),
      fHistMixedPtMult(),
```

a-mathis commented on PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx:
```diff
@@ -285,12 +289,15 @@ void AliSigma0PhotonMotherCuts::SigmaToLambdaGamma(
         fHistInvMassRecLambda->Fill(pT, sigma.GetRecMassLambda());
         fHistInvMassRec->Fill(pT, sigma.GetRecMass());
         if (rapBin > -1) fHistPtY[rapBin]->Fill(pT, invMass);
+        if (multBin == -1) continue;
```
```suggestion
        if (multBin < 0) continue;
```

a-mathis commented on PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx:
```diff
@@ -364,7 +380,10 @@ void AliSigma0PhotonMotherCuts::SigmaToLambdaGammaMixedEvent(
           const int rapBin = GetRapidityBin(rap);
+          const int multBin = GetMultiplicityBin(lPercentile);
           if (rapBin > -1) fHistMixedPtY[rapBin]->Fill(pT, invMass);
+          if (multBin == -1) continue;
```
```suggestion
          if (multBin < 0) continue;
```

a-mathis commented on PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx:
```diff
@@ -393,7 +412,10 @@ void AliSigma0PhotonMotherCuts::SigmaToLambdaGammaMixedEvent(
           const int rapBin = GetRapidityBin(rap);
+          const int multBin = GetMultiplicityBin(lPercentile);
           if (rapBin > -1) fHistMixedPtY[rapBin]->Fill(pT, invMass);
+          if (multBin == -1) continue;
```
```suggestion
          if (multBin < 0) continue;
```

a-mathis commented on PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx:
```diff
@@ -414,11 +436,10 @@ void AliSigma0PhotonMotherCuts::SigmaToLambdaGammaMixedEventBinned(
   const int zVertexBin = GetZvertexBin(zVertex);
   const int multBin = GetMultiplicityBin(lPercentile);
-
-  if (zVertexBin < 0 || multBin < 0) return;
```
Why the change? Was fine before

a-mathis commented on PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx:
```diff
@@ -514,7 +537,8 @@ void AliSigma0PhotonMotherCuts::FillEventBuffer(
   const int zVertexBin = GetZvertexBin(zVertex);
   const int multBin = GetMultiplicityBin(lPercentile);
 
-  if (zVertexBin < 0 || multBin < 0) return;
```
Why the change?

a-mathis commented on PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx:
```diff
@@ -781,6 +824,31 @@ void AliSigma0PhotonMotherCuts::InitCutHistograms(TString appendix) {
 
+  for (int i = 0; i < static_cast<int>(multBins.size() - 1); i++) {
+    fHistPtMult[i] =
+        new TH2F(Form("fHistPtMult_%.2f_%.2f", multBins[i], multBins[i + 1]),
```
```suggestion
        new TH2F(Form("fHistPtMult_%i", i]),
```
To remove the overhead and make the output less chaotic

a-mathis commented on PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx:
```diff
@@ -781,6 +824,31 @@ void AliSigma0PhotonMotherCuts::InitCutHistograms(TString appendix) {
+                 100, 0, 10, 150, 1.15, 1.3);
+    fHistograms->Add(fHistPtMult[i]);
+    fHistMixedPtMult[i] = new TH2F(
+        Form("fHistMixedPtMult%.2f_%.2f", multBins[i], multBins[i + 1]),
```
```suggestion
        Form("fHistMixedPtMult_%i", i),
```

a-mathis commented on PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx:
```diff
@@ -781,6 +824,31 @@ void AliSigma0PhotonMotherCuts::InitCutHistograms(TString appendix) {
+        100, 0, 10, 150, 1.15, 1.3);
+    fHistograms->Add(fHistMixedPtMult[i]);
+    fHistMixedInvMassBinnedMultPt[i] =
+        new TH2F(Form("fHistMixedInvMassBinnedMultPt%.2f_%.2f", multBins[i],
```
```suggestion
        new TH2F(Form("fHistMixedInvMassBinnedMultPt_%i", i
```

a-mathis commented on PWGGA/Hyperon/AliSigma0PhotonMotherCuts.h:
```diff
@@ -129,13 +129,13 @@ class AliSigma0PhotonMotherCuts : public TObject {
-  float fSidebandCutDown;   //
-  float fPhotonPtMin;       //
-  float fPhotonPtMax;       //
-  float fRapidityMax;       //
```
not so nice to just commit empty spaces due to the formatting

makor replied:
I ran the beautifier, should I correct it by hand?

a-mathis replied:
Please

a-mathis replied:
Can you fix this please

a-mathis commented on PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx:
```diff
@@ -393,7 +414,10 @@ void AliSigma0PhotonMotherCuts::SigmaToLambdaGammaMixedEvent(
         fHistMixedInvMassPt->Fill(pT, invMass);
         if (!fIsLightweight) {
```
The histograms are initialised anyways, so could you fill them all the time (not only when fIsLightweight is true

a-mathis commented on PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx:
```diff
@@ -364,7 +382,10 @@ void AliSigma0PhotonMotherCuts::SigmaToLambdaGammaMixedEvent(
         fHistMixedInvMassPt->Fill(pT, invMass);
         if (!fIsLightweight) {
```
The histograms are initialised anyways, so could you fill them all the time (not only when fIsLightweight is true

a-mathis commented on PWGGA/Hyperon/AliSigma0PhotonMotherCuts.cxx:
```diff
@@ -285,12 +291,15 @@ void AliSigma0PhotonMotherCuts::SigmaToLambdaGamma(
       if (std::abs(rap) > fRapidityMax) continue;
       const int rapBin = GetRapidityBin(rap);
+      const int multBin = GetMultiplicityBin(lPercentile);
       if (!fIsLightweight) {
```
The histograms are initialised anyways, so could you fill them all the time (not only when fIsLightweight is true
Just move them to where fHistInvMassPt is filled (and accordingly for the others)

omvazque committed:
Task for rTPC analysis in pp vs mult

omvazque committed:
Taks for rTPC analysis in pp vs mult

fmazzasc committed:
add phi and pt for daughters

fmazzasc committed:
Adding hypertriton structure in task of strangeness

adriansev committed:
Convert AddTaskRhoNew to static method

Conversion of AddTask macros to compile code allow easier usage
and greater compatibility with ROOT6.
The AddTaskRhoNew.C is kept for compatibility.

victor-gonzalez committed:
Fix for wrong pile-up cut assignment in multiplicity range border

ddobrigk committed:
Added first rough OADB for LHC18r

alibuild committed:
Merge pull request #7728 from a-mathis/master

Small improvements for Sigma0 femto

alibuild committed:
Merge pull request #7727 from ddobrigk/master

Added first rough OADB for LHC18r

HubertNowakowski committed:
Fix wrong variable name.

alibuild committed:
Merge pull request #7726 from victor-gonzalez/master

Fix for wrong pile-up cut assignment in multiplicity range border

omvazque committed:
Calibration parameters were included

omvazque committed:
Merge branch 'master' of https://github.com/alisw/AliPhysics

alibuild committed:
Merge pull request #7720 from adriansev/TaskRho_static

Convert AddTaskRhoNew to static method

Commit:
Fix small return issue if AliEventCuts is not used

sartozza committed:
just update some labels in the histos

omvazque committed:
Calibration parameters were included

alibuild committed:
Merge pull request #7730 from omvazque/master

dE/dx vs eta calibration parameters were included

alibuild committed:
Merge pull request #7732 from sartozza/master

just update some labels in the histos

alibuild committed:
Merge pull request #7731 from efranzis/master

Fix small return issue if AliEventCuts is not used

Commit:
Changed Non-Lin parameters. Now for the first time usable

alibuild committed:
Merge pull request #7733 from efranzis/NonLinDev

Changed Non-Lin parameters. Now for the first time usable

fmazzasc committed:
Adding hypertriton structure

fprino committed:
Fix in D+ syst unc

haandrew committed:
Update axis ranges of THnSparse Lund

Commit:
Update AliAnalysisTaskNetLambdaTrad.cxx

alibuild committed:
Merge pull request #7735 from fprino/master

Fix in D+ syst unc

alibuild committed:
Merge pull request #7734 from fmazzasc/master

Adding structure for hypertriton in task of strangeness

alibuild committed:
Merge pull request #7738 from eumaka/ejimaka

Update AliAnalysisTaskNetLambdaTrad.cxx

alibuild committed:
Merge pull request #7737 from haandrew/master

Update axis ranges of THnSparse Lund

alibuild committed:
Merge pull request #7729 from hnowakow/master

Fix wrong variable name in ConfigFemto with corrections.

Commit:
enable to change the cuts values for systematics, add real ID info by missidentified V0 in MC

alibuild committed:
Merge pull request #7739 from lhusova/master

Changes in AliAnalysisTaskDiHadCorrelHighPt

mfasDa committed:
Adding check for rho in the opposite detector

sumitsaha92 committed:
Update AliAnalysisTaskEmcalJetShapeExtra.cxx

fmazzasc committed:
Fix compatibility of selector for the latest task version

Commit:
Changed parameters to effectivley no correction for the other periods. Fixed bug in the parameter set from Nicolas

tomasuzuki committed:
1127_modify

alibuild committed:
Merge pull request #7743 from tomasuzuki/master

1127_modify

alibuild committed:
Merge pull request #7740 from sumitsaha92/master

Update AliAnalysisTaskEmcalJetShapeExtra.cxx

omvazque committed:
Includes MC info to measure efficiencies

fmazzasc committed:
Update linkdef

alibuild committed:
Merge pull request #7745 from omvazque/master

Includes MC info to measure efficiencies

alibuild committed:
Merge pull request #7742 from fmazzasc/master

Fix compatibility of selector for the latest task version

alibuild committed:
Merge pull request #7746 from mfasDa/hotfix-backgroundtask

Adding check for rho in the opposite detector

lvermunt committed:
Fix in Dstar syst unc

alibuild committed:
Merge pull request #7744 from efranzis/master

Changed parameters to effectivley no correction for the other periods…

alibuild committed:
Merge pull request #7747 from lvermunt/master

PWGHF: Fix in Dstar syst unc

nschmidtALICE committed:
PWGGA/GammaConv: Added DMC nonlinearity params for pp 5 TeV

alibuild committed:
Merge pull request #7749 from nschmidtALICE/master

PWGGA/GammaConv: Added DMC nonlinearity params for pp 5 TeV

Commit:
Update AliAnalysisTaskNetLambdaTrad.cxx

Commit:
Update AliAnalysisTaskNetLambdaTrad.h

alibuild committed:
Merge pull request #7750 from eumaka/ejimaka

Ejimaka: update task

lcunquei committed:
more info and again wrong folder

alibuild committed:
Merge pull request #7751 from lcunquei/nov

more info and again wrong folder

fprino committed:
Fix in text box

alibuild committed:
Merge pull request #7752 from fprino/master

Fix in text box

fprino committed:
Add methods needed for the calculation of rebinned xsecs

alibuild committed:
Merge pull request #7753 from fprino/master

Add methods needed for the calculation of rebinned xsecs

srode28 committed:
HFE low B 13 eta 0.5 contamination

efgauger committed:
Added the MC weights for D, B, pi0, and eta for 30-50% centrality. Also added plots to calculate the M20 efficiency both using MC and a data-driven method.

efgauger committed:
Changed the default pi0 and eta weight to 0 if improper centrality range is chosen.

alibuild committed:
Merge pull request #7755 from efgauger/master

Added weights for 30-50% centrality, histos for M20 cut eff.

alibuild committed:
Merge pull request #7754 from srode28/master

HFE low B 13 eta 0.5 contamination

Commit:
Adding conditions on mother and process in Pythia6 MC truth method

sashingo committed:
add a flag for pp analysis

alibuild committed:
Merge pull request #7757 from sashingo/master

add a flag for pp analysis

nzardosh committed:
Added Option for AddTask

alibuild committed:
Merge pull request #7756 from e-masson/master

Adding conditions on mother and process in Pythia6 MC truth method

alibuild committed:
Merge pull request #7758 from nzardosh/master

Added Option for AddTask

dberzano committed:
Do not merge this commit

This is a test commit that fails automatic tests deliberately for the ALICE
Starterkit 2018.

dberzano committed:
Additional change to fix the problem

dberzano committed:
Revert "Additional change to fix the problem"

This reverts commit ad2ffc2cfebf32aef840441f83134987ec3ba8f5.

dberzano committed:
Revert "Do not merge this commit"

This reverts commit ef6c7575f0aa85259a8c62a8457200dd73c8c8fc.
